### PR TITLE
rgw: send headers of quota settings

### PR DESCRIPTION
**Pull request by ofriedma**
Make the quota settings visible to s3 and swift users
rgw send headers of quota settings

Fixes: https://tracker.ceph.com/issues/47752
s3-tests: https://github.com/ceph/s3-tests/pull/371
Signed-off-by: Or Friedmann <ofriedma@redhat.com>

### CODE REVIEW

**Review by dang — APPROVED** [review]

File: src/rgw/rgw_rest_s3.cc [hunk 1]
```diff
@@ -2140,6 +2140,14 @@ static void dump_bucket_metadata(struct req_state *s, rgw::sal::RGWBucket* bucke
 {
   dump_header(s, "X-RGW-Object-Count", static_cast<long long>(bucket->get_count()));
   dump_header(s, "X-RGW-Bytes-Used", static_cast<long long>(bucket->get_size()));
+  // only bucket's owner is allowed to get the quota settings of the account
+  if (bucket->is_owner(s->user.get())) {
+    auto user_info = s->user->get_info();
+    dump_header(s, "X-RGW-Quota-Bucket-size", static_cast<long long>(user_info.bucket_quota.max_size));
```
Comment (mattbenjamin) [inline src/rgw/rgw_rest_s3.cc:2146]: capitalization doesn't match the existing headers

**Comment by ofriedma** [comment]
need to change it to capital letter

**Comment by ofriedma** [comment]
jenkins test make check

**Comment by ofriedma** [comment]
jenkins test docs

**Comment by ofriedma** [comment]
jenkins test make check

**Comment by ofriedma** [comment]
I Added: 
1) quota info to getusage api and fixed it (it caused segfault in master):
https://docs.ceph.com/en/latest/radosgw/s3/serviceops/#get-usage-stats
2) max buckets header to head bucket op
3) Added head bucket header printing to boto3 examples
@dang @mattbenjamin Can you please take a look again?

**Review by dang — APPROVED** [review]

**Comment by ofriedma** [comment]
http://pulpito.ceph.com/ofriedma-2020-10-19_19:14:20-rgw-wip-ofriedma-show-quota-stats-i01-distro-basic-smithi/5539497/

**Comment by ofriedma** [comment]
@dang @mattbenjamin Do you think we can merge it?

**Review by mattbenjamin — APPROVED** [review]
lgtm; looks like teuthology run is a pass

**Pull request merged**